sympy/__init__.py

```diff
@@ -496,3 +496,171 @@ def __sympy_debug():
     # sympy.deprecated:
     'C', 'ClassRegistry', 'class_registry',
 ]
+
+
+#=======================================================================
+#
+# XXX: The names below were importable before sympy 1.6 using
+#
+#          from sympy import *
+#
+# This happened implicitly because there was no __all__ defined in this
+# __init__.py file. Not every package is imported. The list matches what
+# would have been imported before.
+#
+#=======================================================================
+
+
+__all__.extend([
+    'algebras',
+    'assumptions',
+    'calculus',
+    'codegen',
+    'combinatorics',
+    'concrete',
+    'deprecated',
+    'discrete',
+    'external',
+    'functions',
+    'geometry',
+    'interactive',
+    'multipledispatch',
+    'ntheory',
+    'parsing',
+    # 'physics',
+    'plotting',
+    'polys',
+    'printing',
+    'release',
+    'strategies',
+    'tensor',
+    'utilities',
+])
+
+
+#=======================================================================
+#
+# XXX: The names below were importable before sympy 1.6 using
+#
+#          from sympy import *
+#
+# This happened implicitly because there was no __all__ defined in this
+# __init__.py file. The plan is to remove them but for now they remain
+# importable but will give a deprecation warning when used.
+#
+#=======================================================================
+
+
+class DeprecatedImportModule:
```

Comment:
This should subclass types.ModuleType.

Reply:
What difference does subclassing ModuleType make?

Reply:
It makes certain things that operate on modules work correctly. For example, `help()` gives special output on modules (actually that is wrong anyway, because this doesn't inherit the docstring properly). 

I also wonder if this could be an issue

```py
>>> from sympy import ode as ode1
>>> from sympy.solvers import ode as ode2
>>> ode1 == ode2
False
```

It might or might not be an issue, I'm not sure. I guess if this turns out to cause problems, we can do a patch release that removes it.

Reply:
I think I'd prefer the help to be a message saying that you haven't imported the module in the right way (if there is any need for help at all).

The purpose of providing these deprecated module imports is just to keep code that possibly depends on importing this way chugging along but with warnings. I don't see any need to provide any other functionality.